rc/tools/go/gopls.kak

```diff
@@ -0,0 +1,116 @@
+# gopls.kak: gopls bindings for kakoune
+
+define-command -params 1 -docstring %{
+gopls <command>: gopls command wrapper for kakoune
+
+All commands are forwarded to gopls utility
+Available commands are:
+    format
+    imports
+    definition
+    references
+} -shell-script-candidates %{
+    printf "format\nimports\ndefinition\nreferences\n"
+} \
+gopls %{
+    require-module gopls
+    evaluate-commands %sh{
+        case "$1" in
+        format|imports)
+            printf %s\\n "gopls-cmd $1"
+            ;;
+        definition)
+            printf %s\\n "gopls-def"
+            ;;
+        references)
+            printf %s\\n "gopls-ref"
+            ;;
+        *)
+            printf "fail Unknown gopls command '%s'\n" "$1"
+            exit
+            ;;
+        esac
+    }
+}
+
+provide-module gopls %§
+
+evaluate-commands %sh{
+    if ! command -v gopls > /dev/null 2>&1; then
+        echo "fail Please install gopls or add to PATH!"
+    fi
+}
+
+# Temp dir prepatation
+declare-option -hidden str gopls_tmp_dir
+define-command -hidden -params 0 gopls-prepare %{
+    evaluate-commands %sh{
+        dir=$(mktemp -d "${TMPDIR:-/tmp}"/kak-gopls.XXXXXXXX)
+        printf %s\\n "set-option buffer gopls_tmp_dir ${dir}"
+    }
+}
+
+# gopls format/imports
+define-command -hidden -params 1 gopls-cmd %{
+    gopls-prepare
+    evaluate-commands %sh{
+        dir=${kak_opt_gopls_tmp_dir}
+        eval "gopls $1 -w ${kak_buffile} 2> '${dir}/stderr'"
+        if [ $? -ne 0 ]; then
+            # show error messages in *debug* buffer
+            printf %s\\n "echo -debug %file{ '${dir}/stderr' }"
+        fi
+        rm -r "${dir}"
+    }
+    edit!
+}
+
+# gopls definition
+define-command -hidden -params 0 gopls-def %{
+    gopls-prepare
+    evaluate-commands %sh{
+        dir=${kak_opt_gopls_tmp_dir}
+        eval "gopls definition ${kak_buffile}:${kak_cursor_line}:${kak_cursor_column} \
+            |sed -e 's/-.*//' -e 's/:/ /g' > '${dir}/jump'"
```

Comment:
`gopls` will give an output like

```
/usr/lib/go/src/fmt/print.go:212:6-12: defined here as func fmt.Printf(format string, a ...interface{}) (n int, err error)
Printf formats according to a format specifier and writes to standard output.
It returns the number of bytes written and any write error encountered.
```

we only want to pick the first line, for example by using `sed 's/-.*//; s/:/ /g; q'`